src/World_TEST.cc

```diff
@@ -691,3 +707,113 @@ TEST(DOMWorld, Plugins)
   world.ClearPlugins();
   EXPECT_TRUE(world.Plugins().empty());
 }
+
+/////////////////////////////////////////////////
+TEST(DOMWorld, LoadEarthSurface)
+{
+  std::string sdf =
+    "<?xml version='1.0'?>"
+    "<sdf version='1.10'>"
+    "<world name='spherical coordinates'>"
+    "<spherical_coordinates>"
+    " <surface_model>EARTH_WGS84</surface_model>"
+    "  <world_frame_orientation>ENU</world_frame_orientation>"
+    "  <latitude_deg>-22.9</latitude_deg>"
+    "  <longitude_deg>-43.2</longitude_deg>"
+    "  <elevation>0</elevation>"
+    "  <heading_deg>0</heading_deg>"
+    "</spherical_coordinates>"
+    "</world>"
+    "</sdf>";
+
+  sdf::Root root;
+  sdf::World world;
+  root.LoadSdfString(sdf);
```

Comment:
you can just get the sdf errors when loading the Root. I'd also recommend adding expectations on the spherical coordinates values from the C++ API.

```suggestion
  auto sdfErrors = root.LoadSdfString(sdf);
  EXPECT_TRUE(sdfErrors.empty());

  sdf::World *world = root.WorldByIndex(0);
  auto sc = world->SphericalCoordinates();
  EXPECT_NEAR(-22.9, sc.LatitudeReference().Degree(), 1e-6);
  // and other expectations
```

Reply:
Done https://github.com/gazebosim/sdformat/pull/1050/commits/cae46b44a39eaf95d0f3df9a887013255c2cdd5c